core/README.md

```diff
@@ -109,9 +109,36 @@ implementations that cater to the specific requirements of their applications, t
 scalability. The ability to create specialized implementations for specific use cases, coupled with the framework's
 openness to user-defined extensions, makes it a versatile and customizable solution.
 
+#### Understanding Sync and Async Ports
+
+This section explains the differences between `Sync` and `Async` ports.
+
+There are two types of constraints used to calculate the number of input and output samples to be processed:
+
+1. **Port Constraints**: Minimum and maximum number of samples set individually for each port.
+2. **Resampling Constraints**: Block-level settings such as `input_chunk_size`, `output_chunk_size`, and `stride`.
+
+#### Behavior of Sync and Async Ports
+
+**Sync Ports:**
+
+- All `Sync` ports process the same number of samples in each `processBulk()` call.
+- There is a fixed input-to-output sample ratio of `M:N`. The `processBulk()` function is invoked with `k*M`
+  input samples and produces `k*N` output samples.
+- Both port constraints and resampling constraints are applied to calculate the common number of samples to process.
+
+**Async Ports:**
+
+- If at least one port is attributed as `Async`, the `processBulk()` function is called if at least one `Async` port
+  meets its port constraints, regardless of the sample availability on other ports.
+- Only port constraints are applied; there is no predefined input/output sample ratio.
+- The block is responsible for manually consuming and publishing samples and tags.
+- Input tags are consumed within the range of the consumed samples unless the block explicitly calls
+  `consumeTags()` inside `processBulk()`.
```

Comment:
Thanks a lot for your comment! I’ve updated the documentation as you suggested. I haven’t included the parts about tags yet, but I think we should add them after we complete the Tag Forward Policies. Since it’s a big topic, it would fit nicely in the section related to tags.